src/test/run-make/obtain-borrowck/driver.rs

```diff
@@ -0,0 +1,171 @@
+#![feature(rustc_private)]
+
+//! This program implements a rustc driver that retrieves MIR bodies with
+//! borrowck information. This cannot be done in a straightforward way because
+//! `get_body_with_borrowck_facts`–the function for retrieving a MIR body with
+//! borrowck facts–can panic if the body is stolen before it is invoked.
+//! Therefore, the driver overrides `mir_borrowck` query (this is done in the
+//! `config` callback), which retrieves the body that is about to be borrow
+//! checked and stores it in a thread local `MIR_BODIES`. Then, `after_analysis`
+//! callback triggers borrow checking of all MIR bodies by retrieving
+//! `optimized_mir` and pulls out the MIR bodies with the borrowck information
+//! from the thread local storage.
+
+extern crate rustc_driver;
```

Comment:
This test looks like it's in the wrong directory -- any test using compiler crates should be in run-make-fulldeps. I think that'll also mean you can get rid of the BUILD_STAGE and -L flags etc that you're using above.

You can look at https://github.com/rust-lang/rust/blob/master/src/test/run-make-fulldeps/hotplug_codegen_backend/Makefile and other files in that directory for a similar test to copy off of.

Reply:
Thanks! I thought that I am missing something… I will try moving it.